syntax/grammar.mjs

```diff
@@ -86,12 +86,15 @@ let CommentLine = defer(() =>
 
 /* ------------------------------------------------------------------------- */
 /* Adjacent junk_lines are joined into FTL.Junk during the AST construction. */
-let junk_line = defer(() =>
+let junk_line =
     sequence(
         regex(/.*/),
-        line_end)
+        either(
+            string("\u000A"),
+            string("\u000D\u000A"),
```

Comment:
I wonder if it would make sense to add other, technically unsupported, line endings here. In particular, I'm thinking of `\r`, `\u2028` and `\u2029`, which are not matched by `.` in the JavaScript regex implementation.

We don't need to include [all line terminators recognized by Unicode](https://en.wikipedia.org/wiki/Newline#Representation), but adding these extra three would make sure the reference parser can produce meaningful junk for files with these endings. Otherwise, the parser doesn't have any way to skip over the EOL in the `junk_line` production, and the parsing of the whole resource ends immediately.

Reply:
Would `regex(/[^\n]*/`) be closer to our intent? And not change `line_end` at all?

Reply:
This is a really good idea! Thanks.

I had some trouble making it work when I tested it. I was seeing additional `\n` where I was sure there shouldn't be any. And then I realized that I needed to use `echo -n` for my CLI testing...